### PR TITLE
change create-react-class to javascript class for to reduce the size …

### PR DESCRIPTION
**Pull request by ilyjs**
…of the source file

### CODE REVIEW

File: src/chartsFactory.jsx [hunk 1]
```diff
@@ -1,79 +1,88 @@
...
+
+    class Charts extends Component {
+        constructor() {
+            super()
+            this.chartType = chartType;
+            this.Highcharts = Highcharts;
+            this.displayName = 'Highcharts' + chartType;
+        }
+
+        setChartRef = (chartRef) => {
```
Comment (kirjs) [inline src/chartsFactory.jsx]: How does this work?
Change to setChartRef(chartRef) ?

File: src/chartsFactory.jsx [hunk 2]
```diff
@@ -1,79 +1,88 @@
...
-        win && win.requestAnimationFrame && requestAnimationFrame(()=>{
-          this.chart && this.chart.options && this.chart.reflow();
-        });
-      }
-    },
+        renderChart(config) {
+            if (!config) {
+                throw new Error('Config must be specified for the ' + this.displayName + ' component');
+            }
+            let chartConfig = config.chart;
```
Comment (kirjs) [inline src/chartsFactory.jsx]: const?

File: src/chartsFactory.jsx [hunk 3]
```diff
@@ -1,79 +1,90 @@
...
+        componentWillUnmount() {
+            this.chart.destroy();
+        }
+
+        render() {
+            return <div ref={this.setChartRef.bind(this)} {...this.props.domProps} />;
+        }
+    }
+
+    if (cutCode) {
```
Comment (kirjs) [inline src/chartsFactory.jsx]: rename to isProdMode or something similar?

File: src/chartsFactory.jsx [hunk 4]
```diff
@@ -1,79 +1,90 @@
...
-      }
-      let chartConfig = config.chart;
-      this.chart = new Highcharts[chartType]({
-        ...config,
-        chart: {
-          ...chartConfig,
-          renderTo: this.chartRef
+function chartsFactory(chartType, Highcharts) {
+
+    class Charts extends Component {
```
Comment (kirjs) [inline src/chartsFactory.jsx]: Just chart?

Comment (ilyjs) [reply src/chartsFactory.jsx]: Yes)

File: src/chartsFactory.jsx (hunk 3 above)
Comment (ilyjs) [reply src/chartsFactory.jsx]: Ok

**Pull request merged**